cql3/query_processor.cc

```diff
@@ -1025,16 +1026,28 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
 
     cql3::cql_warnings_vec warnings;
 
+    auto request_id = guard->new_group0_state_id();
+    const_cast<statements::schema_altering_statement&>(stmt).global_req_id = request_id;
+
     auto [ret, m, cql_warnings] = co_await stmt.prepare_schema_mutations(*this, options, guard->write_timestamp());
     warnings = std::move(cql_warnings);
 
+    ce = std::move(ret);
     if (!m.empty()) {
         auto description = format("CQL DDL statement: \"{}\"", stmt.raw_cql_statement);
-        co_await remote_.get().mm.announce(std::move(m), std::move(*guard), description);
+        if (ce && ce->target == cql_transport::event::schema_change::target_type::TABLET_KEYSPACE) {
+            co_await remote_.get().mm.announce<service::topology_change>(std::move(m), std::move(*guard), description);
```

Comment:
So it looks like you are explicitly calling `announce` of a certain type. so why bother making it a template instead of just adding a function so ss that sends the request and waits for it? It does not make sense to send `topology_change` mutation through mm.

Reply:
Do you mean that `mm` is meant to create only schema mutations? OK, I haven't thought about it this way.
I'll add this to the list of potential follow-ups.

Reply:
Yes. Migration manager is schema related. Note that you have other ss function calls in this `if` branch and you even add one of them to ss, so it is better to move everything to ss.

Reply:
Added to https://github.com/scylladb/scylladb/issues/19126